### DISCUSSION TITLE
docs: how to install and run a simple VM from iso

### DISCUSSION THREAD

**Discussion by hw-claudio** [cloud-hypervisor/cloud-hypervisor#5529] [Q&A]
Beyond saying that cloud hypervisor is a rust vmm,
the documentation is not good at providing answers to very basic questions about cloud-hypervisor features, and how it is supposed to be used.

There is a lot of "you can do this, you can do that", but nothing to show what cloud-hypervisor is actually capable of, and what are its limitations.

Is it possible to take a CentOS7 iso and run its installation, then run the CentOS7 VM?

The only example available talks about supplying specific kernels, fimwares, preparing a cloud-init .. and a bunch more stuff.

How does one do something as basic as installing and running CentOS7? Thanks.

**Comment by likebreath** [comment]
No, Cloud Hypervisor does not support ISO images nor support installation from it. 

It is generally expected to use provisioned images with Cloud Hypervisor, mostly commonly cloud images. We have a list of supported/tested cloud images that is basically Ubuntu Focal and Jammy, and Fedora 36 [1]. I believe image provisioning is out of the scope of the project.

[1] https://github.com/cloud-hypervisor/cloud-hypervisor#3-status

**Comment by hw-claudio** [comment]
Hi,

if you don't consider the case of booting an iso directly, and you don't plan any integration with something like virt-install either, I think you should document that, as it can generate confusion.

As stated before I don't think that the project page explains clearly what cloud-hypervisor can do and what it cannot do.

This specific choice makes this VMM arbitrarily limited by the way; with qemu still necessary to pre-prepare the images, one can at that point continue using qemu. I suppose then that the procedure that is required currently to run Windows is not just a temporary thing, it is really how you expect this to be used, which is a bit absurd but your choice of course.
